python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -1160,9 +1138,17 @@ def _create_ufind(self, optimize_ops):
                     ufind.union(op1, op2)
         return ufind
 
-    def _is_opt_op(self, op):
-        # NOTE: It's a HACK implement.
-        # optimize op: SGDOptimize, MomentumOptimizer, AdamOptimizer and etc...
+    def _is_opt_role_op(self, op):
+        # NOTE: depend on oprole to find out whether this op is for
+        # optimize
+        op_maker = core.op_proto_and_checker_maker
+        optimize_role = core.op_proto_and_checker_maker.OpRole.Optimize
+        if op_maker.kOpRoleAttrName() in op.attrs and \
+            int(op.attrs[op_maker.kOpRoleAttrName()]) == int(optimize_role):
+            return True
+        return False
+
+    def _is_optimizer_op(self, op):
```

Comment:
Maybe we can use `op_role` to determinate whether the op is a **optimize** one.

Reply:
ops like weight decay and cliping and other ops will marked as optimize role. The other function `_is_opt_role_op` will do this job.

Reply:
Got it, thanks!